yang.go

```diff
@@ -202,6 +204,14 @@ Formats:
 			names = append(names, m.Name)
 		}
 	}
+	if f.includeSubmodules {
+		for _, m := range ms.SubModules {
+			if mods[m.Name] == nil {
```

Comment:
can `m` ever be nil at this point?  If it could be, perhaps the branch should instead be:

```go
if m != nil && mods[m.Name] == nil {
  // existing branch body goes here
}
```

Reason being that if `m` were nil, the branch could be executed twice, leading to `names` having a duplicate in it.

(I think `m` probably cannot be nil; if you're sure that's the case, ignore this comment)